truffleruby+graalvm-dev fails to download for install

### Discussed in https://github.com/rbenv/ruby-build/discussions/2242

<div type='discussions-op-text'>

<sup>Originally posted by **halostatue** September  1, 2023</sup>
```console
$ ruby-build truffleruby+graalvm-dev ~/.rubies/truffleruby+graalvm-dev
To follow progress, use 'tail -f /var/folders/c6/k_j3j3w11590l8x6t8rwxw200000gn/T/ruby-build.20230901120615.95886.log' or pass --verbose
ruby-build: using openssl from homebrew
Downloading graalvm-community-java17-darwin-aarch64-dev.tar.gz...
-> https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-darwin-aarch64-dev.tar.gz
error: failed to download truffleruby+graalvm-dev.tar.gz
```

This is affecting any GitHub Action which is using `truffleruby+graalvm-head` (which uses `truffleruby+graalvm-dev`):

```
  /home/runner/work/_temp/ruby-build-for-setup-ruby/bin/ruby-build truffleruby+graalvm-dev /home/runner/.rubies/truffleruby+graalvm-head
  To follow progress, use 'tail -f /tmp/ruby-build.20230901153346.1773.log' or pass --verbose
  Downloading graalvm-community-java17-linux-amd64-dev.tar.gz...
  -> https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-linux-amd64-dev.tar.gz
  error: failed to download truffleruby+graalvm-dev.tar.gz
  
  BUILD FAILED (Ubuntu 20.04 using ruby-build 20230717-11-g4d4678b)
```

See https://github.com/KineticCafe/app_identity/actions/runs/6051304100/job/16422454037?pr=69

I have not opened an issue with ruby/setup-ruby because this is reproducible with ruby-build on my macOS, not just the GitHub Action runner using Ubuntu 20.04 or 22.04.</div>

--- a/share/ruby-build/truffleruby+graalvm-dev
+++ b/share/ruby-build/truffleruby+graalvm-dev
@@ -1,18 +1,18 @@
 platform="$(uname -s)-$(uname -m)"
 case $platform in
 Linux-x86_64)
-  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-linux-amd64-dev.tar.gz" truffleruby_graalvm
+  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java21-linux-amd64-dev.tar.gz" truffleruby_graalvm
   ;;
 Linux-aarch64)
-  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-linux-aarch64-dev.tar.gz" truffleruby_graalvm
+  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java21-linux-aarch64-dev.tar.gz" truffleruby_graalvm
   ;;
 Darwin-x86_64)
   use_homebrew_openssl
-  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-darwin-amd64-dev.tar.gz" truffleruby_graalvm
+  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java21-darwin-amd64-dev.tar.gz" truffleruby_graalvm
   ;;
 Darwin-arm64)
   use_homebrew_openssl
-  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-darwin-aarch64-dev.tar.gz" truffleruby_graalvm
+  install_package "truffleruby+graalvm-dev" "https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java21-darwin-aarch64-dev.tar.gz" truffleruby_graalvm
   ;;
 *)
   colorize 1 "Unsupported platform: $platform"

